numerical/area/bounds/_base.py

```diff
@@ -0,0 +1,101 @@
+import numpy as np
+from abc import ABCMeta, abstractmethod
+import matplotlib.pyplot as plt
+
+
+class BaseBoundary(metaclass=ABCMeta):
+    @abstractmethod
+    def plot(self, save_path=None):
+        """ Visualizes boundary. """
+
+    def _set_described_rect(self):
+        """ Set n-dim rectangular bound which describes initial bound. """
+
+    @abstractmethod
+    def _is_boundary_valid(self):
+        """ Validate the data which gives for boundary initialization. """
+
+
+class BoundaryIterable(metaclass=ABCMeta):
+    """ The class represents iterator interface for boundaries. """
+    def __init__(self, data: list):
+        self._data = data
+        self.bounds_count = len(data)
+
+    def __iter__(self):
+        self._bound_id = 0
+        return self
+
+    def __next__(self):
+        if self._bound_id == self.bounds_count:
+            raise StopIteration
+        self._bound_id += 1
+        return self._data[self._bound_id - 1]
+
+    def __getitem__(self, item):
+        return self._data[item]
+
+    def __repr__(self):
+        return "<[" + ", ".join([bnd.__repr__() for bnd in self._data]) + "]>"
+
+
+class LineBoundary(BaseBoundary):
+    """ The class represents start and end of 1-dim line. """
+    def __init__(self, start, end):
+        self.start = np.float64(start)
+        self.end = np.float64(end)
+
+        if not self._is_boundary_valid():
+            raise BoundarySetupException
+
+    def _is_boundary_valid(self):
+        return self.end > self.start and not np.allclose(self.start, self.end)
+
+    def plot(self, save_path=None):
+        plt.hlines(0.0, self.start, self.end, colors='black')
+        plt.axvline(self.start, color='black', linestyle='--')
+        plt.axvline(self.end, color='black', linestyle='--')
+        if save_path:
+            plt.savefig(save_path)
+        else:
+            plt.show()
+
+    def __repr__(self):
+        return f"<{self.__class__.__name__}: start={self.start} end={self.end}>"
+
+
+class PolarRhoLineBoundary(LineBoundary):
+    """ The class represents start and end of 1-dim line
```

Comment:
Polar coordiates are two-dimensional AFAIK, no?

Reply:
I use these two classes (PolarRhoLineBoundary, PolarPhiLineBoundary) to create boundaries in polar coordinates.